lib/src/front_end/delimited_list_builder.dart

```diff
@@ -387,32 +389,88 @@ class DelimitedListBuilder {
     );
   }
 
-  /// If [_blockCandidates] contains a single expression that can receive
-  /// block formatting, then returns its index. Otherwise returns `-1`.
-  int _findBlockElement() {
+  /// Looks at the [BlockFormat] types of all of the elements to determine if
+  /// one of them should be block formatted.
+  ///
+  /// Also, if an argument list has an adjacent strings expression followed by a
+  /// block formattable function expression, we allow the adjacent strings to
+  /// split without forcing the list to split so that it can continue to have
+  /// block formatting. This is pretty special-cased, but it makes calls to
+  /// `test()` and `group()` look better and those are so common that it's
+  /// worth massaging them some. It allows:
+  ///
+  ///     test('some long description'
+  ///         'split across multiple lines', () {
+  ///       expect(1, 1);
+  ///     });
+  ///
+  /// Without this special rule, the newline in the adjacent strings would
+  /// prevent block formatting and lead to the entire test body to be indented:
+  ///
+  ///     test(
+  ///       'some long description'
+  ///       'split across multiple lines',
+  ///       () {
+  ///         expect(1, 1);
+  ///       },
+  ///     );
+  ///
+  /// Stores the result of this calculation by setting flags on the
+  /// [ListElements].
+  void _selectBlockElement() {
     // TODO(tall): These heuristics will probably need some iteration.
     var functions = <int>[];
     var others = <int>[];
+    var adjacentStrings = <int>[];
 
     for (var i = 0; i < _elements.length; i++) {
       switch (_elements[i].blockFormat) {
         case BlockFormat.function:
           functions.add(i);
         case BlockFormat.block:
           others.add(i);
+        case BlockFormat.indentedAdjacentStrings:
+        case BlockFormat.unindentedAdjacentStrings:
+          adjacentStrings.add(i);
         case BlockFormat.none:
           break; // Not a block element.
       }
     }
 
     // A function expression takes precedence over other block arguments.
-    if (functions.length == 1) return functions.first;
+    if (functions.length == 1) {
```

Comment:
[optional] 
```suggestion
    if (functions case [var singleFunction]) {
```
I think the name `singleFunction` will be more informative than `functions.first` below.

Reply:
Ooh, good idea. Went all the way:

```dart
    switch ((functions, others, adjacentStrings)) {
      case ([1], _, [0]):
        // The adjacent strings.
        _elements[0].allowNewlines = true;
        if (_elements[0].blockFormat == BlockFormat.unindentedAdjacentStrings) {
          _elements[0].indentWhenBlockFormatted = true;
        }

        // The block-formattable function.
        _elements[1].allowNewlines = true;

      // A function expression takes precedence over other block arguments.
      case ([var blockArgument], _, _):
      // Otherwise, if there is single block argument, it can be block formatted.
      case ([], [var blockArgument], _):
        _elements[blockArgument].allowNewlines = true;
    }
```